allow versioned symlinks

jhheider opened the pull request:
- [x] support semver ranges on `tea -X`
- [x] support semver ranges on symlinks
- [x] support links to semver ranged symlinks (yup, really: `node > node^16 > tea`)
- [x] expand symlink README documentation to cover new feature
- [x] implement test for new feature

jhheider commented:
Ugh. Typo in the readme. Need to reverse ln arguments in the second invocation in the script. Tomorrow.

mxcl commented:
Don’t see the harm in it and probably will be considered a nice feature by many. Though we should document that `python3` == `python^3` (we already support that).

mxcl commented on README.md:
````diff
@@ -189,6 +189,18 @@ Notably if you create a symlink of `foo` to `tea` (or `tea_foo`) we will
+
+```sh
+$ ln -s tea node^16
+$ ln -s node node^16
````
aaah nice thinking!

mxcl commented on tests/integration/tea-XX.test.ts:
```diff
@@ -23,3 +23,10 @@ Deno.test("tea -SX python3.11", async () => {
   })
 })
+
+Deno.test("tea -SX node^16", async () => {
```
solid addition

jhheider commented:
> Though we should document that `python3` == `python^3` (we already support that).

Added that, and cleaned up my typo, as well as using explicit paths to avoid confusion.

mxcl commented:
Wonderful additions Jacob
